indexer/borders.cpp

```diff
@@ -0,0 +1,95 @@
+#include "indexer/borders.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/geometry_coding.hpp"
+#include "coding/var_record_reader.hpp"
+
+#include "base/logging.hpp"
+#include "base/macros.hpp"
+
+#include "defines.hpp"
+
+using namespace std;
+
+namespace
+{
+template <class ReaderT>
+class BordersVector
+{
+  DISALLOW_COPY(BordersVector);
+
+public:
+  BordersVector(ReaderT const & reader) : m_recordReader(reader, 256 /* expectedRecordSize */) {}
+
+  template <class ToDo>
+  void ForEach(ToDo && toDo) const
+  {
+    m_recordReader.ForEachRecord([&](uint32_t pos, char const * data, uint32_t /*size*/) {
+      ArrayByteSource src(data);
+      serial::GeometryCodingParams cp = {};
+      uint64_t id;
+      ReadPrimitiveFromSource(src, id);
+      size_t size;
+      ReadPrimitiveFromSource(src, size);
+      size_t outerSize;
+      ReadPrimitiveFromSource(src, outerSize);
+      vector<m2::PointD> outer(outerSize);
+      // todo:(@t.yan) consider delta coding techniques.
+      for (size_t i = 0; i < outer.size(); ++i)
```

Comment:
nit: if you don't use reserve+emplace_back, you can write range-based for here instead

Reply:
done